### PR TITLE
[DOC] Adding documentation to xlnet scripts

### PR DESCRIPTION
**Pull request by zburning**
## Description ##
Adding doc to xlnet scripts

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-985/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-985/1/index.html

**Comment by mli** [comment]
Job PR-985/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-985/2/index.html

**Comment by leezu** [comment]
Thanks @zburning! The linting step is still currently still failing (see http://ci.mxnet.io/blue/organizations/jenkins/GluonNLP-py3-master-gpu-integration/detail/PR-985/2/pipeline#step-68-log-1133 )
Could you help to fix it?

For running lint locally, execute `pylint --rcfile .pylintrc  PATH_TO_FILES_TO_CHECK`. To install all optional and development dependencies of GluonNLP, you can run `cd GLUONNLP; pip install -e '.[extras,dev]'`. The development dependencies include everything needed to run the linting.

**Comment by mli** [comment]
Job PR-985/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-985/3/index.html

File: scripts/language_model/transformer/transformer.py [hunk 1]
```diff
@@ -486,6 +486,45 @@ def hybrid_forward(self, F, inputs, pos_emb, mem_value, mask, segments):
...
+    use_decoder : bool, default True
+        Whether to include the decoder for language model prediction.
+    tie_decoder_weight : bool, default True
+        Whether to tie the decoder weight with the input embeddings
+    weight_initializer : str or Initializer
+        Initializer for the input weights matrix, used for the linear
+        transformation of the inputs.
+    bias_initializer : str or Initializer
+        Initializer for the bias vector.
+    prefix : str, default 'rnn_'
```
Comment (leezu) [inline scripts/language_model/transformer/transformer.py]: default is `None`

File: scripts/language_model/transformer/transformer.py [hunk 2]
```diff
@@ -486,6 +486,45 @@ def hybrid_forward(self, F, inputs, pos_emb, mem_value, mask, segments):
...
+        Prefix for name of `Block`s (and name of weight if params is `None`).
+    params : Parameter or None
```
Comment (leezu) [inline scripts/language_model/transformer/transformer.py]: Should be ParameterDict or None

**Comment by mli** [comment]
Job PR-985/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-985/4/index.html

**Comment by mli** [comment]
Job PR-985/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-985/5/index.html

**Pull request merged**